### ISSUE TITLE
Customer support and SOS

### ISSUE THREAD

**Issue by ashwingattani** [p2prider/p2prider#13]
This is coming from a personal experience, where once I took an auto, and when the driver started smoking bidi, I asked him not to. Upon which he started verbally abusing me. I took this auto from Ola and reported them but I am not sure if anything happened to the driver. We would need a support system for users and drivers where they can report/rate the ride and help us filter out spam.

**Comment by arjunhm** [comment]
Customer support may not be feasible in this case, but a rating system could help deter both parties from abusing the system.  
The rating system can be simple where both parties are given the option to rate the person at the end of the ride. They are also shown the other person's rating prior to accepting the ride. It is up to the them to decide if they want to go ahead with it.  

One possible issue is that the drivers can increase their ratings if it is low.  
Ex: Driver A has a rating of 2/5. The driver and his/her friends can create accounts as riders and create rides with Driver A (of course they can skip the payment part since the app is not collecting any money). Now they are given the option to rate Driver A, and they go ahead and give a 5-star rating.

**Comment by ashwingattani** [comment]
I think ratings with [drivers marked as fav](https://github.com/p2prider/p2prider/issues/11#issue-1174319835) as a combination can help us achieve the true goal.

Additionally, we can have an SOS button in the application, which can help users click and end the ride in-between. On clicking SOS, the driver can be rated negative (I am not sure if this should be in MVP) and can be tackled accordingly.

**Comment by vvkishere** [comment]
I think there are a few things we can do as optional. I am just brain storming here so rejecting these can be a possibility. 
1. Keep an auto recording feature for both the customer and the driver to start immediately when the trip has started - this can be both Audio and video. We can create a free streaming to a specific video site of their choice that can be deleted upon completion of the trip if unnecessary. The recording can be watermarked specifically by our program to ensure authenticity. 
2. The driver and customer can also stream their trip to a favourite user of their choice to keep track of them when necessary. 
3. Make an optional plug in for anyone willing to conduct this service for a fee that they can run the necessary authentications for the cabs or customers. Our app can be the core for an ecosystem of service providers that help everyone.

**Comment by ashwingattani** [comment]
@vvkishere for proper streaming, we would need dashcams on all registered vehicles. I am not sure if that would be feasible as of now.

**Comment by papasierragolf** [comment]
Also, VBR would be needed to be able to stream on all sort of networks etc. 
I think, we can track the locations of both the driver and rider and then if there's any discrepancy, we don't show the rating option. Tracking can also be done, with bluetooth of each device to figure out that they are in the same vehicle. 
As in, if the driver even gets his friends to do it, after 2-3 rides, he will be burning his own money. 
Also, we put in rating system for different length of rides - as in 1-2 km ride has a different priority of rating than a 25kms ride. Just as in real life, we may judge someone at first instance, but as we get to slowly know them, we realise they are actually better than what we thought. 
So basically what im saying is, if its a shorter ride, we build in that logic, that the rating of that ride, doesn't affect the overall driver / rider rating as much as a longer ride would effect.

**Comment by papasierragolf** [comment]
Also, would we be able to have a live dashboard on the backend to 

1. track all cabs and live rides
2. track specific user who has raised a flag / alert and hand it over to the authorities / alert local police etc?
